docs/source/features/hydra.rst

```diff
@@ -0,0 +1,122 @@
+Hydra Configuration System
+==========================
+
+.. currentmodule:: omni.isaac.lab
+
+Isaac Lab supports the `Hydra <https://hydra.cc/docs/intro/>`_ configuration system to modify the task's
+configuration using command line arguments, which can be useful to automate experiments and perform hyperparameter tuning.
+
+Any parameter of the environment can be modified by adding one or multiple elements of the form ``env.a.b.param1=value``
+to the command line input, where ``a.b.param1`` reflects the parameter's hierarchy, for example ``env.actions.joint_effort.scale=10.0``.
+Similarly, the agent's parameters can be modified by using the ``agent`` prefix, for example ``agent.seed=2024``.
+
+.. note::
+
+    The way these command line arguments are set follow the exact structure of the configuration files. Since the different
+    RL frameworks use different conventions, there might be differences in the way the parameters are set. For example,
+    with `rl_games` the seed will be set with ``agent.params.seed``, while with `rsl_rl` and `skrl` it will be set with
+    ``agent.seed``.
+
+
+As a result, training with hydra arguments can be run with the following syntax:
+
+.. tab-set::
+    :sync-group: rl-train
+
+    .. tab-item:: rsl_rl
+        :sync: rsl_rl
+
+        .. code-block:: shell
+
+            python source/standalone/workflows/rsl_rl/train.py --task=Isaac-Cartpole-v0 --headless env.actions.joint_effort.scale=10.0 agent.seed=2024
+
+    .. tab-item:: rl_games
+        :sync: rl_games
+
+        .. code-block:: shell
+
+            python source/standalone/workflows/rl_games/train.py --task=Isaac-Cartpole-v0 --headless env.actions.joint_effort.scale=10.0 agent.params.seed=2024
+
+    .. tab-item:: skrl
+        :sync: skrl
+
+        .. code-block:: shell
+
+            python source/standalone/workflows/skrl/train.py --task=Isaac-Cartpole-v0 --headless env.actions.joint_effort.scale=10.0 agent.seed=2024
+
+The above command will run the training script with the task ``Isaac-Cartpole-v0`` in headless mode, and set the
+``env.actions.joint_effort.scale`` parameter to 10.0 and the ``agent.seed`` parameter to 2024.
+
+.. note::
+
+    To keep backwards compatibility, and to provide a more user-friendly experience, we have kept the old cli arguments
+    of the form ``--param``, for example ``--num_envs``, ``--seed``, ``--max_iterations``. These arguments have precedence
+    over the hydra arguments, and will overwrite the values set by the hydra arguments.
+
+.. attention::
+
+    Particular care should be taken when modifying the parameters using command line arguments. Some of the configurations
```

Comment:
This can be addressed using OmegaConf resolvers if we want to provide a "hydra-friendly" patch to the configs. I can follow up here with a simple example.